unittests/test_loader.py

```diff
@@ -73,9 +74,88 @@ def test_load_bad_init(self):
+            class TestDeprecated(rfm.RegressionTest):
+                # Should raise a warning
+                def setup(self, partition, environ, **job_opts):
+                    super().setup(system, environ, **job_opts)
+
+            @rfm.simple_test
+            class TestDeprecatedRunOnly(rfm.RunOnlyRegressionTest):
+                # Should raise a warning
+                def setup(self, partition, environ, **job_opts):
+                    super().setup(system, environ, **job_opts)
+
+            @rfm.simple_test
+            class TestDeprecatedCompileOnly(rfm.CompileOnlyRegressionTest):
+                # Should raise a warning
+                def setup(self, partition, environ, **job_opts):
+                    super().setup(system, environ, **job_opts)
+
+            @rfm.simple_test
+            class TestDeprecatedCompileOnlyDerived(TestDeprecatedCompileOnly):
+                # Should not raise a warning because the setup of the parent
+                # was not set as final
+                def setup(self, partition, environ, **job_opts):
+                    super().setup(system, environ, **job_opts)
+
+            @rfm.simple_test
+            class TestExtended(rfm.RegressionTest, extended_test=True):
+                def __init__(self):
+                    pass
+
+                # Should not raise a warning
+                def setup(self, partition, environ, **job_opts):
+                    super().setup(system, environ, **job_opts)
+
+            @rfm.simple_test
+            class TestExtendedDerived(TestExtended):
+                def __init__(self):
+                    pass
+
+                # Should not raise a warning
+                def setup(self, partition, environ, **job_opts):
+                    super().setup(system, environ, **job_opts)
+
+                # Should raise a warning
+                def run(self):
+                    super().run()
+
+            @rfm.simple_test
+            class TestExtendedRunOnly(rfm.RunOnlyRegressionTest,
+                                      extended_test=True):
+                def __init__(self):
+                    pass
+
+                # Should not raise a warning
+                def setup(self, partition, environ, **job_opts):
+                    super().setup(system, environ, **job_opts)
+
+                # Should not raise a warning
+                def run(self):
+                    super().run()
+
+            @rfm.simple_test
+            class TestExtendedCompileOnly(rfm.CompileOnlyRegressionTest,
+                                          extended_test=True):
+                def __init__(self):
+                    pass
+
+                # Should not raise a warning
+                def setup(self, partition, environ, **job_opts):
+                    super().setup(system, environ, **job_opts)
+
+                # Should not raise a warning
+                def run(self):
+                    super().run()
+
+        assert len(record) == 4
```

Comment:
This is not enough, because you don't have control over what was actually raised.